cmd/util/api.go

```diff
@@ -79,26 +67,68 @@ func GetAPIClientV2(cmd *cobra.Command, cfg types.Bacalhau) (clientv2.API, error
 		clientv2.WithHeaders(headers),
 	}
 
-	authTokenPath, err := cfg.AuthTokensPath()
+	var resolvedAuthToken *apimodels.HTTPCredential
+
+	// Check if the credentials are valid
+	apiKeyOrBasicAuthFlowEnabled, credentialScheme, credentialString, err := resolveAuthCredentials(
+		apiAuthAPIKey,
+		basicAuthUsername,
+		basicAuthPassword,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("authentication error: %v", err)
+	}
+
+	legacyAuthTokenFilePath, err := cfg.AuthTokensPath()
 	if err != nil {
 		log.Warn().Err(err).Msg("Failed to read access tokens path – API calls will be without authorization")
 	}
-	existingAuthToken, err := ReadToken(authTokenPath, base)
+
+	// Try to get the tokens file for SSO tokens
+	ssoAuthTokenPath, err := cfg.JWTTokensPath()
+	if err != nil {
+		log.Warn().Err(err).Msg("Failed to read access jwt tokens path")
+	}
+
+	// Do not error out if we are not able to do that , just log
+	existingSSOCredential, err := ReadTokenFn(ssoAuthTokenPath, baseURL)
 	if err != nil {
-		log.Warn().Err(err).Msg("Failed to read access tokens – API calls will be without authorization")
+		log.Warn().Err(err).Msg("Failed to read SSO access tokens file")
 	}
 
+	// If credentials are provided, add them to the headers
+	if apiKeyOrBasicAuthFlowEnabled {
+		resolvedAuthToken = &apimodels.HTTPCredential{
+			Scheme: credentialScheme,
+			Value:  credentialString,
+		}
+		log.Debug().Msg("Using API Key or Basic Auth authentication credentials")
+	} else if existingSSOCredential != nil {
+		resolvedAuthToken = existingSSOCredential
+		log.Debug().Msg("Using SSO authentication credentials")
+	} else {
+		resolvedAuthToken, err = ReadTokenFn(legacyAuthTokenFilePath, baseURL)
+		if err != nil {
+			log.Warn().Err(err).Msg("Failed to read access tokens – API calls will be without authorization")
+		}
+	}
+
+	newAuthenticationFlowEnabled := apiKeyOrBasicAuthFlowEnabled || existingSSOCredential != nil
+	skipAuthentication := cmd.Use == "sso" || cmd.Use == "version" || cmd.Use == "alive"
+
 	userKeyPath, err := cfg.UserKeyPath()
 	if err != nil {
 		return nil, err
 	}
 
 	return clientv2.NewAPI(
 		&clientv2.AuthenticatingClient{
-			Client:     clientv2.NewHTTPClient(base, opts...),
-			Credential: existingAuthToken,
+			Client:                       clientv2.NewHTTPClient(baseURL, opts...),
+			Credential:                   resolvedAuthToken,
+			NewAuthenticationFlowEnabled: newAuthenticationFlowEnabled,
+			SkipAuthentication:           skipAuthentication,
```

Comment:
nit: New will be old at one point :) 
but its fine as I think this is temporary and will be removed in next release when we drop support for legacy auth

Reply:
Todo